cosmos/chain/chain.go

```diff
@@ -361,42 +361,45 @@ func (c *Chain) Init(ctx context.Context, opts petritypes.ChainOptions) error {
 		}
 	}
 
-	peers, err := c.PeerStrings(ctx)
-	if err != nil {
-		return err
+	chainConfig := c.GetConfig()
+	var persistentPeers, seeds string
+
+	if chainConfig.SetPersistentPeers {
+		persistentPeers, err = PeerStrings(ctx, append(c.Nodes, c.Validators...))
+		if err != nil {
+			return err
+		}
+	}
+
+	if chainConfig.SetSeedNode {
+		var seedNode petritypes.NodeI
+		if len(c.Nodes) > 0 {
+			seedNode = c.Nodes[0]
+		} else {
+			seedNode = c.Validators[0]
```

Comment:
You can't use the regular nodes as a seed. They need to have the proper config set https://github.com/cometbft/cometbft/blob/9ccdb9bb68324c48f088857b45ac2fabfa9879c4/config/config.go#L594 

This will cause them to connect, share peers, then disconnect.